### ISSUE TITLE
projectService / EXPERIMENTAL_useProjectService gives TS error if not boolean on v8 alpha / v7

### ISSUE THREAD

**Issue by hkatzdev** [typescript-eslint/typescript-eslint#9313]
### Before You File a Bug Report Please Confirm You Have Done The Following...

- [X] I have tried restarting my IDE and the issue persists.
- [X] I have updated to the latest version of the packages.
- [X] I have [searched for related issues](https://github.com/typescript-eslint/typescript-eslint/issues?q=is%3Aissue+label%3Abug) and found none that matched my issue.
- [X] I have [read the FAQ](https://typescript-eslint.io/linting/troubleshooting) and my problem is not listed.

### Relevant Package

types

### ESLint Config

```javascript
import tseslint from 'typescript-eslint';

// v8 alpha
export default tseslint.config(
  {
    languageOptions: {
      parserOptions: {
        projectService: {
          allowDefaultProject: ['./*.js'],
          defaultProject: './tsconfig.json',
        },
      },
    },
  },
);

// v7
export default tseslint.config(
  {
    languageOptions: {
      parserOptions: {
        EXPERIMENTAL_useProjectService: {
          allowDefaultProject: ['./*.js'],
          defaultProject: './tsconfig.json',
        },
      },
    },
  },
);
```

### tsconfig

```jsonc
{
  "compilerOptions": {
    "allowJs": true,
    "checkJs": true,
  }
}
```

### Expected Result

Config should not have a type error.

### Actual Result

On line 7, there is an error.
On v8 alpha:
```
Type '{ allowDefaultProject: string[]; defaultProject: string; }' is not assignable to type 'boolean'.ts(2322)
parser-options.d.ts(30, 5): The expected type comes from property 'projectService' which is declared here on type 'ParserOptions'
```
On v7:
```
Type '{ allowDefaultProject: string[]; defaultProject: string; }' is not assignable to type 'boolean'.ts(2322)
parser-options.d.ts(25, 5): The expected type comes from property 'EXPERIMENTAL_useProjectService' which is declared here on type 'ParserOptions'
```

### Additional Info

Config is adapted from https://typescript-eslint.io/blog/announcing-typescript-eslint-v8-beta/#project-service.
At a glance, it seems like there is a type mismatch between https://github.com/typescript-eslint/typescript-eslint/blob/3c3370fc1f0ae2d4ecd8e91bfc5fdb01dba9fc17/packages/typescript-estree/src/parser-options.ts#L220 and https://github.com/typescript-eslint/typescript-eslint/blob/3c3370fc1f0ae2d4ecd8e91bfc5fdb01dba9fc17/packages/types/src/parser-options.ts#L70. It looks like a similar issue exists in the stable version (https://github.com/typescript-eslint/typescript-eslint/blob/63e53e28f68f84764ec632d502da3d33095f6a0d/packages/typescript-estree/src/parser-options.ts#L141 and https://github.com/typescript-eslint/typescript-eslint/blob/63e53e28f68f84764ec632d502da3d33095f6a0d/packages/types/src/parser-options.ts#L65)

### Versions

v8 alpha
| package                                | version |
| -------------------------------------- | ------- |
| `@typescript-eslint/types`     | `8.0.0-alpha.29` |
| `@typescript-eslint/parser`    | `8.0.0-alpha.29` |
| `typescript-eslint`                 | `8.0.0-alpha.29` |

v7
| package                                | version |
| -------------------------------------- | ------- |
| `@typescript-eslint/types`     | `7.12.0` |
| `@typescript-eslint/parser`    | `7.12.0` |
| `typescript-eslint`                 | `7.12.0` |

**Comment by bradzacher** [comment]
Cc @JoshuaKGoldberg
